src/circleci/rollcage/ring_middleware.clj

```diff
@@ -1,12 +1,32 @@
 (ns circleci.rollcage.ring-middleware
-  (:require [circleci.rollcage.core :as rollcage]))
-
-(defn wrap-rollbar [handler rollcage-client]
-  (if-not rollcage-client
-    handler
-    (fn [req]
-      (try
-        (handler req)
-        (catch Exception e
-          (rollcage/error rollcage-client e {:url (:uri req)})
-          (throw e))))))
+  (:require
+   [clojure.string :as str]
+   [circleci.rollcage :as rollcage]))
+
+(def ^:dynamic *request-scope* (atom {}))
+
+(defn add-to-custom!
+  "Add custom data m into the data that will be sent to rollbar."
+  [fields]
+  (swap! *request-scope* update :custom merge fields))
+
+(defn add-to-request!
+  "Add request data m into the data that will be sent to rollbar."
+  [fields]
+  (swap! *request-scope* update :request merge fields))
+
+(defn add-to-person!
+  "Add user data m into the data that will be sent to rollbar."
+  [fields]
+  (swap! *request-scope* update :person merge fields))
+
+(defn wrap-rollbar [handler]
```

Comment:
If this middleware is used in a context where there's a default exception handler installed which uses Rollbar then exceptions will end up twice in Rollbar. In general, one should not send to Rollbar and then rethrow. Rollbar should be the last thing done before either handling the exception or exiting the thread.